pkg/exporter/process.go

```diff
@@ -44,7 +44,7 @@ type ExportingProcess struct {
 	obsDomainID     uint32
 	seqNumber       uint32
 	templateID      uint16
-	set             *entities.Set
+	set             *entities.BaseSet
```

Comment:
This should be interface, so that it can be applied for both DataSet and TemplateSet. In addition, BaseSet should be embedded in DataSet and TemplateSet--they both should implement the inerface Set. Do you think that works?

Reply:
Agreed that it should be interface and I will need to refactor set structure. I kept the original idea from you to use one set struct to templateSet and dataSet to make things more tidy and reusable. Thanks!